packages/application-shell/src/test-utils/test-utils.js

```diff
@@ -116,20 +120,28 @@ const render = (
   const mergedEnvironment = mergeOptional(defaultEnvironment, environment);
   return {
     ...rtlRender(
-      <IntlProvider locale={locale}>
-        <ApolloProviderComponent mocks={mocks} addTypename={addTypename}>
-          <ConfigureFlopFlip adapter={adpater} defaultFlags={flags}>
-            <ApplicationContextProvider
-              user={mergedUser}
-              project={mergedProject && { ...mergedProject, permissions }}
-              environment={mergedEnvironment}
-              projectDataLocale={dataLocale}
-            >
-              <Router history={history}>{ui}</Router>
-            </ApplicationContextProvider>
-          </ConfigureFlopFlip>
-        </ApolloProviderComponent>
-      </IntlProvider>,
+      <div>
+        <IntlProvider locale={locale}>
+          <ApolloProviderComponent mocks={mocks} addTypename={addTypename}>
+            <ConfigureFlopFlip adapter={adpater} defaultFlags={flags}>
+              <ApplicationContextProvider
+                user={mergedUser}
+                project={mergedProject && { ...mergedProject, permissions }}
+                environment={mergedEnvironment}
+                projectDataLocale={dataLocale}
+              >
+                <Router history={history}>
+                  <React.Suspense fallback={<LoadingFallback />}>
+                    {ui}
+                  </React.Suspense>
+                </Router>
+              </ApplicationContextProvider>
+            </ConfigureFlopFlip>
+          </ApolloProviderComponent>
+        </IntlProvider>
+        {/* This container is used to render modals, so we need to make it available in tests */}
+        <div id={PORTALS_CONTAINER_ID} />
```

Comment:
And this is also new, which is required to render the modals etc

Reply:
I've been thinking about this some more. I'm not sure our current solution is the best one. It seems a bit weird to always provide the portal container (or maybe I'm just not used to it) - thereby messing with the actual container DOM for every test case. All our other setup functions don't influence the container's DOM tree itself.

The way it's done in `react-testing-library`'s course is that they expose a `Modal` which you can render into your tests: see [example](https://github.com/kentcdodds/react-testing-library-course/blob/master/src/__tests__/portals.js#L11-L13). We could expose that `Modal` component from our test-utils. This is nice because it's declarative and doesn't mess with the test cases in which we don't need a modal.

But maybe it's a better idea to always render the portal container as we're doing now. It's nice that devs don't need to worry about the modal at all if we always put it in place. I'm really not sure, and both approaches obviously work. I'm torn between them. It's up to you which way to go.

Maybe let's proceed with what we have here, and in case it gets annoying/confusing we can try the other approach.

Reply:
I agree with your thoughts in general, however in our case the AppShell always renders it and therefore I think it should be provided by default as well since most of our views use the modal 😊

Reply:
I'm totally aware of that :)

What's giving me discomfort is that we call `render(<Test />)`, but the container will contain not just the output of `Test`. I think this can be quite confusing for first-timers. I dislike this disconnect. But maybe it's the best decision to take. Anyways, let's proceed as-is 🚴